js/src/components/paid-ads/budget-section/index.js

```diff
@@ -32,36 +35,56 @@ const nonInteractableProps = {
  * @param {Array<CountryCode>|undefined} props.countryCodes Country codes to fetch budget recommendations for.
  * @param {boolean} [props.disabled=false] Whether display the Card in disabled style.
  * @param {JSX.Element} [props.children] Extra content to be rendered under the card of budget inputs.
+ * @param {'create-ads'|'edit-ads'|'setup-ads'|'setup-mc'} props.context A context indicating which page this component is used on.
  */
 const BudgetSection = ( {
 	formProps,
 	countryCodes,
 	disabled = false,
 	children,
+	context,
 } ) => {
-	const { getInputProps, setValue, values } = formProps;
+	const { getInputProps, values, setValue } = formProps;
 	const { amount } = values;
 	const { googleAdsAccount } = useGoogleAdsAccount();
+	const {
+		data: budgetRecommendationData,
+		highestDailyBudget,
+		highestDailyBudgetCountryCode,
+		hasFinishedResolution,
+	} = useFetchBudgetRecommendation( countryCodes );
```

Comment:
One of the solutions I suggested for resolving the issue of https://github.com/woocommerce/google-listings-and-ads/pull/2552#discussion_r1804597699 is to adjust the timing of issuing the recommended budget API request, and invalidate the corresponding states in the wp-data store after disconnecting Google Ads account.

The current implementation is in this direction but it's not ideal. This implementation might only be considered when there is no relatively suitable and easier one.

Suggestions:

1. Revert these commits: https://github.com/woocommerce/google-listings-and-ads/compare/48aeec3..f4631e9
2. Adjust the structure of components under `js/src/setup-ads` to a similar one as [js/src/setup-mc/setup-stepper/setup-paid-ads.js](https://github.com/woocommerce/google-listings-and-ads/blob/48aeec39d95edc688bece8ad072ba7d37a4aa5fc/js/src/setup-mc/setup-stepper/setup-paid-ads.js)
   - Move `CampaignAssetsForm` and all related codes from `SetupAdsForm` to `SetupPaidAds`
   - Move `SetupAdsTopBar` and `AdsStepper` from `SetupAdsForm` to `SetupAds`
   - After the above changes, `SetupAdsForm` component should be empty and can be removed
3. It should only need to invalidate the recommended budget states when after disconnecting Google Ads account
   https://github.com/woocommerce/google-listings-and-ads/blob/f4631e95b90458f0e07b7cc278c90624797a75b7/js/src/data/resolvers.js#L565-L570

Reply:
Thanks @eason9487 . I think your suggestions make sense. Updated.